Changing the status of MCG in MTC (and OADP) documentation (#78361)
openshift/openshift-docs: modules/migration-configuring-mcg.adoc

= Retrieving Multicloud Object Gateway credentials

ifdef::installing-3-4,installing-mtc[]

> You must retrieve the Multicloud Object Gateway (MCG) credentials and S3 endpoint to configure MCG as a replication repository for the {mtc-full} ({mtc-short})

Reviewer: I might leave _in order_ here.  It makes it more clear.  Now you have to stop for a moment to think:

 _endpoint to configure_  Are you retrieving the endpoint that you want to configure or retrieving the endpoint so that you can use it to configure something else?

Reviewer: @RichardHoch I just looked in the IBM Style Guide. It suggested using _to_ over _in order to_ "in most cases" to avoid wordiness. Based on what stillalearner replies (I would think the former is right), how about:

```suggestion
You must retrieve the Multicloud Object Gateway (MCG) credentials and S3 endpoint, which you need to configure MCG as a replication repository for the {mtc-full} ({mtc-short})
```

Author: @mburke5678 A fine solution, pending QA's answer. :)

Author: @mburke5678  QA confirmed my reading of the discussion of the S3 endpoint. Made the change you suggested.

> You must retrieve the Multicloud Object Gateway (MCG) credentials to create a `Secret` custom resource (CR) for {mtc-short}.

Reviewer: Same thing here. 
```suggestion
You must retrieve the Multicloud Object Gateway (MCG) credentials in order to create a `Secret` custom resource (CR) for {mtc-short}.
```

Reviewer 2: Unresolving this as I see its still needing to be addressed or a reason why not. @RichardHoch I'll let you tell me when you're ready for merge after you address this.

Author: @skrthomas You're right, I missed this. Sorry!

endif::[]

ifdef::installing-oadp-mcg[]

> You must retrieve the Multicloud Object Gateway (MCG) credentials to create a `Secret` custom resource (CR) for the OpenShift API for Data Protection (OADP).

Reviewer: ```suggestion
You must retrieve the Multicloud Object Gateway (MCG) credentials in order to create a `Secret` custom resource (CR) for {mtc-short}.
```

Author: @mburke5678 This is the text for OADP. There's a separate text for MTC.

Reviewer: @RichardHoch Sorry! I cut and pasted my [suggestion from the previous](https://github.com/openshift/openshift-docs/pull/78361#pullrequestreview-2180828308).  This is what I intended:

```suggestion
You must retrieve the Multicloud Object Gateway (MCG) credentials in order to create a `Secret` custom resource (CR) for the OpenShift API for Data Protection (OADP).
```

endif::[]
//ifdef::installing-oadp-mcg[]
//endif::[]

[NOTE]
====

> Although the MCG Operator was link:https://catalog.redhat.com/software/containers/ocs4/mcg-rhel8-operator/5ddbcefbdd19c71643b56ce9?architecture=amd64&image=64243f5dcd0eb61355af9abd[deprecated], the MCG plugin is still available for {rh-storage}. To download the plugin, browse to link:https://access.redhat.com/downloads/content/547/ver=4/rhel---9/4.15.4/x86_64/product-software[Download {rh-storage-first}] and download the appropriate MCG plugin for your operating system.

Reviewer: _is deprecated_ is more common in our docs than _was deprecated_. The latter seem to be used with a specific version or event (_the feature was deprecated in version 7.23_)
```suggestion
Although the MCG Operator is link:https://catalog.redhat.com/software/containers/ocs4/mcg-rhel8-operator/5ddbcefbdd19c71643b56ce9?architecture=amd64&image=64243f5dcd0eb61355af9abd[deprecated], the MCG plugin is still available for {rh-storage}. To download the plugin, browse to link:https://access.redhat.com/downloads/content/547/ver=4/rhel---9/4.15.4/x86_64/product-software[Download {rh-storage-first}] and download the appropriate MCG plugin for your operating system.
```

====

.Prerequisites
ifdef::openshift-origin[]
* Ensure that you have downloaded the {cluster-manager-url-pull} as shown in _Obtaining the installation program_ in the installation documentation for your platform.
+
If you have the pull secret, add the `redhat-operators` catalog to the OperatorHub custom resource (CR) as shown in _Configuring {product-title} to use Red Hat Operators_.
endif::[]
* You must deploy {rh-storage} by using the appropriate link:https://docs.redhat.com/en/documentation/red_hat_openshift_data_foundation/4.15[Red Hat OpenShift Data Foundation deployment guide].

> .Procedure

Author: Note for reviewers: For MTC, this is a one-step procedure, which is supposed to use an asterisk, rather than a number, for the step. For OADP this is a two-step procedure, which is supposed to use numbers for the steps. I tried to use a snippet for the step "Obtain the S3 endpoint..." to avoid writing it in each "if" statement, but I could not get the * or steps' numbers to format properly. So I have left the step in each "if" statement.  If there is a way to avoid this repetition, please let me know.

ifdef::installing-3-4,installing-mtc[]

> * Obtain the S3 endpoint, `AWS_ACCESS_KEY_ID`, and `AWS_SECRET_ACCESS_KEY` by running the link:https://access.redhat.com/documentation/en-us/red_hat_openshift_data_foundation/4.15/html/managing_hybrid_and_multicloud_resources/accessing-the-multicloud-object-gateway-with-your-applications_rhodf#accessing-the-Multicloud-object-gateway-from-the-terminal_rhodf[`describe` command] on the `NooBaa` custom resource.

Reviewer: I reckon we should be updating links to access.redhat to the new docs.redhat.com 
```suggestion
* Obtain the S3 endpoint, `AWS_ACCESS_KEY_ID`, and `AWS_SECRET_ACCESS_KEY` by running the link:https://docs.redhat.com/en/documentation/red_hat_openshift_data_foundation/4.15/html/managing_hybrid_and_multicloud_resources/accessing-the-multicloud-object-gateway-with-your-applications_rhodf#accessing-the-Multicloud-object-gateway-from-the-terminal_rhodf`describe` command] on the `NooBaa` custom resource.
```
